__fixtures__/pnpm-workspace.yaml

```diff
@@ -18,3 +18,7 @@ packages:
   - '!workspace-has-shared-package-lock-json'
   - '!workspace-has-shared-npm-shrinkwrap-json'
 sharedWorkspaceLockfile: false
+
+catalog:
+  # Used in has-outdated-deps-using-catalog-protocol fixture.
+  is-negative: ^1.0.0
```

Comment:
This had to be added since we're installing the `has-outdated-deps-using-catalog-protocol` fixture's dependencies using this `pnpm-workspace.yaml` on CI.

https://github.com/pnpm/pnpm/blob/ad0cfad1b859f5f2a707828c7a6061f86e4cc59c/__fixtures__/package.yaml#L2

The bug fix revealed a place we were accidentally relying on the old (buggy) behavior.